### DISCUSSION TITLE
Trying to dump flash for a Nano Every (ATMEGA4809)

### DISCUSSION THREAD

**Discussion by 641i130** [avrdudes/avrdude#1356] [Q&A] [answered]
I'm currently trying to dump the firmware for my 'nano every' but I keep getting 'programmer is not responding'.

```avrdude -p m4809 -c avr910 -P/dev/ttyACM0 -U flash:r:"out.hex":r```

How should I go about fixing my command?

**Comment by MCUdude** [comment]
If you're using the on-board USB connector, you can't use `avr910` as the programmer, you'll have to use `jtag2updi`.

so:

```
avrdude -p m4809 -c jtag2updi -P/dev/ttyACM0 -U flash:r:"out.hex":r
```

However, the Nano every requires a "1200bps touch" in order to enter jtag2updi mode instead of the typical serial port.
This means that a serial port is opened at 1200bps, and then closed again before running Avrdude. On my mac, I've been using `stty` to do this, but it might be different on your computer.

```
$ PORT=/dev/ttyACM0; stty -f $PORT 1200 && avrdude -pm4808 -cjtag2updi -P$PORT -Uflash:r:out.hex:r
```

**Reply by 641i130** [reply]
Ahh, that makes sense. Thank you!

Is it normal to see:
```
avrdude warning: attempt 1 of 10: sign-on command: status -1
avrdude warning: attempt 2 of 10: sign-on command: status -1
avrdude warning: attempt 3 of 10: sign-on command: status -1
avrdude warning: attempt 4 of 10: sign-on command: status -1
avrdude warning: attempt 5 of 10: sign-on command: status -1
avrdude warning: attempt 6 of 10: sign-on command: status -1
avrdude warning: attempt 7 of 10: sign-on command: status -1
avrdude warning: attempt 8 of 10: sign-on command: status -1
avrdude warning: attempt 9 of 10: sign-on command: status -1
avrdude warning: attempt 10 of 10: sign-on command: status -1
avrdude error: timeout/error communicating with programmer (status -1)
avrdude error: unable to open programmer jtag2updi on port /dev/ttyACM0

avrdude done.  Thank you.
```

Can you provide the stty version or website you got it off of? I don't think my Linux distro supports it.

**Reply by MCUdude** [reply]
Sorry, but the stty version that's bundled with macOS doesn't support -h, --help, -v or --version, but this is what I get when running the `what` command:
```
$ what /bin/stty
/bin/stty
	PROGRAM:stty  PROJECT:adv_cmds-172
```
However, I'm sure @mcuee has used stty for the exact same purpose, but on a computer running Linux. Maybe he can tell?

Anyways, here's what a working command should look like when connected to a genuine Nano Every:

```
$ PORT=/dev/cu.usbmodem14101; stty -f $PORT 1200 && ./avrdude -pm4809 -cjtag2updi -P$PORT -Uflash:r:out.hex:r
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9651 (probably m4809)
avrdude: reading flash memory ...
Reading | ################################################## | 100% 5.22 s 
avrdude: writing output file out.hex

avrdude done.  Thank you.
```

**Reply by 641i130** [reply, marked as answer]
AH! I figured it out. My version of `stty` must be newer:
```PORT=/dev/ttyACM0; stty -F $PORT 1200 && avrdude -pm4809 -cjtag2updi -P$PORT -Uflash:r:out.hex:r ```
Thank you very much!

**Reply by MCUdude** [reply]
Great to hear you figured it out!

**Reply by mcuee** [reply]
More discussions for different OS here: Windows is the most troublesome.
* https://github.com/avrdudes/avrdude/issues/788

**Reply by MCUdude** [reply]
@mcuee we should probably gather various "tricks" like this in a Wiki page to make it easier to find for others that might run into the same problem(s).

**Reply by mcuee** [reply]
And yes for Linux. the correct option is to use `-F`.
https://man7.org/linux/man-pages/man1/stty.1.html

**Reply by mcuee** [reply]
For macOS, it is using `-f`.
https://www.unix.com/man-page/osx/1/stty/

**Reply by mcuee** [reply]
> @mcuee we should probably gather various "tricks" like this in a Wiki page to make it easier to find for others that might run into the same problem(s).

I have created the following page and added a few FAQ items.
https://github.com/avrdudes/avrdude/wiki/FAQ

**Reply by mcuee** [reply]
@MCUdude 

Please feel free to change the formating and add more items.
